dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/DatadogCore.kt

```diff
@@ -173,20 +177,21 @@ internal class DatadogCore(
     }
 
     /** @inheritDoc */
+    @AnyThread
     override fun addUserProperties(extraInfo: Map<String, Any?>) {
         coreFeature.userInfoProvider.addUserProperties(extraInfo)
     }
 
     /** @inheritDoc */
+    @AnyThread
     override fun clearAllData() {
         features.values.forEach {
             it.clearAllData()
         }
-        // TODO RUM-1462 address Thread safety
-        @Suppress("ThreadSafety") // removal of the data is done in synchronous manner
-        coreFeature.deleteLastViewEvent()
-        @Suppress("ThreadSafety") // removal of the data is done in synchronous manner
-        coreFeature.deleteLastFatalAnrSent()
+        getPersistenceExecutorService().submitSafe("Clear all data", internalLogger) {
+            coreFeature.deleteLastViewEvent()
+            coreFeature.deleteLastFatalAnrSent()
+        }
     }
```

Comment:
I think it is done in sync way by design, so that the call on the caller side is blocking (even though it may happen on the main thread) and the same ideology as with `features.values.forEach { it.clearAllData() }` above. Is it blocking in iOS SDK?

Reply:
Just checked, on iOS the call is async